### PR TITLE
Add GCC compiler switch: -mfsmuld for SPARC.

### PR DESCRIPTION
**Pull request by klausz65**

### CODE REVIEW

**Comment by AndWac** [comment]
This changes obviously more than just the introduction of a new folder constant.

File: make-rules/shared-macros.mk [hunk 1]
```diff
@@ -240,9 +240,10 @@ USRLIBDIR64 =	$(USRDIR)/lib/$(MACH64)
 USRSHAREDIR =	$(USRDIR)/share
 USRINCDIR = 	$(USRDIR)/include
 USRSHARELOCALEDIR =	$(USRSHAREDIR)/locale
-USRSHAREMANDIR =	$(USRSHAREDIR)/man
```
Comment (mtelka) [inline make-rules/shared-macros.mk]: Why this change?  These lines are _not_ sorted alphabetically.

File: make-rules/shared-macros.mk [hunk 2]
```diff
@@ -267,7 +268,7 @@ PROTOETCDIR =	$(PROTO_DIR)/$(ETCDIR)
 PROTOETCSECDIR = $(PROTO_DIR)/$(ETCDIR)/security
 PROTOUSRDIR =	$(PROTO_DIR)/$(USRDIR)
 PROTOBINDIR =	$(PROTO_DIR)/$(BINDIR)
-PROTOSBINDIR =	$(PROTO_DIR)/$(SBINDIR)
+PROTOSBINDIR =	$(PROTO_DIR)$(USRDIR)$(SBINDIR)
```
Comment (mtelka) [inline make-rules/shared-macros.mk]: Huh.  What is this?  Why?

File: make-rules/shared-macros.mk [hunk 3]
```diff
@@ -240,9 +240,10 @@ USRLIBDIR64 =	$(USRDIR)/lib/$(MACH64)
 USRSHAREDIR =	$(USRDIR)/share
 USRINCDIR = 	$(USRDIR)/include
 USRSHARELOCALEDIR =	$(USRSHAREDIR)/locale
-USRSHAREMANDIR =	$(USRSHAREDIR)/man
 USRSHAREDOCDIR =	$(USRSHAREDIR)/doc
+USRSHAREHELPDIR =	$(USRSHAREDIR)/help
```
Comment (mtelka) [inline make-rules/shared-macros.mk]: Why is this new directory needed?

File: make-rules/shared-macros.mk [hunk 4]
```diff
@@ -284,9 +285,10 @@ PROTOUSRLIBDIR32 =	$(PROTO_DIR)/$(USRLIBDIR32)
 PROTOUSRLIBDIR64 =	$(PROTO_DIR)/$(USRLIBDIR64)
 PROTOUSRINCDIR =	$(PROTO_DIR)/$(USRINCDIR)
 PROTOUSRSHAREDIR =	$(PROTO_DIR)/$(USRSHAREDIR)
+PROTOUSRSHAREDOCDIR =	$(PROTO_DIR)/$(USRSHAREDOCDIR)
+PROTOUSRSHAREHELPDIR =	$(PROTO_DIR)/$(USRSHAREHELPDIR)
 PROTOUSRSHARELIBDIR =	$(PROTO_DIR)/$(USRSHARELIBDIR)
 PROTOUSRSHAREMANDIR =	$(PROTO_DIR)/$(USRSHAREMANDIR)
-PROTOUSRSHAREDOCDIR =	$(PROTO_DIR)/$(USRSHAREDOCDIR)
```
Comment (mtelka) [inline make-rules/shared-macros.mk]: Why this change?  These lines are _not_ sorted alphabetically.

**Review by mtelka — CHANGES_REQUESTED** [review]

File: make-rules/shared-macros.mk (hunk 1 above)
Comment (mtelka) [reply make-rules/shared-macros.mk]: ... they are not sorted and they are not intended to be sorted.

**Comment by mtelka** [comment]
I suggest to split this PR to two changes: one for sparc settings and the other one for directories.  It looks like you are trying to do two unrelated things in one changeset.  Thanks.

**Comment by klausz65** [comment]
Split this PR into several ones.

**Pull request closed without merging**